watertap/tools/oli_api/flash.py

```diff
@@ -75,420 +73,1175 @@
+            requests_to_process,
+            # burst_job_tag=burst_job_tag,
+            # max_concurrent_processes=max_concurrent_processes,
+            # batch_size=batch_size,
+        )
         _logger.info("Completed running flash calculations")
+        result = flatten_results(processed_requests)
         if file_name:
-            self.write_output(result, file_name)
-
+            write_output(result, file_name)
         return result
 
-    def modify_inputs(self, initial_flash_input, survey, flash_method):
+    def get_clone(self, flash_method, json_input, index, survey=None):
         """
-        Iterates over a survey to create modified clones of an initial flash analysis output.
+        Iterate over a survey to create a modified clone from JSON input.
 
-        :param initial_flash_input: flash analysis input to copy
-        :param survey: dictionary containing modifications for each test
-        :param flash_method: string name of flash method to use
+        :param flash_method: string for flash calculation name
+        :param json_input: JSON input for flash calculation
+        :param index: integer for index of incoming data
+        :param survey: dictionary containing names and input values to modify in JSON
 
-        :return clones: dictionary containing modified state variables and survey index
+        :return clone: dictionary containing modified state variables and survey index
         """
 
-        clones = {}
-        for i in survey.keys():
-            modified_clone = deepcopy(initial_flash_input)
+        if survey is None:
+            return json_input
+
+        valid_flashes = [
+            "wateranalysis",
+            "isothermal",
+            "isenthalpic",
+            "bubblepoint",
+            "dewpoint",
+            "vapor-amount",
+            "vapor-fraction",
+            "isochoric",
+            "setph",
+            "precipitation-point",
+            "corrosion-rates",
+        ]
+        if flash_method not in valid_flashes:
+            raise RuntimeError(
+                "Invalid flash_method: {flash_method}. Use one of {', '.join(valid_flashes)}"
+            )
+
+        clone = deepcopy(json_input)
+        for k, v in survey.items():
+            d = clone["params"]
             if flash_method == "wateranalysis":
-                for param in modified_clone["params"]["waterAnalysisInputs"]:
-                    if param["name"] in survey[i].keys():
-                        param.update({"value": survey[i][param["name"]]})
-            elif flash_method == "isothermal":
-                for param in survey[i]:
-                    if param in ["Temperature", "Pressure"]:
-                        modified_clone["params"][param.lower()]["value"] = survey[i][
-                            param
-                        ]
-                    elif param in modified_clone["params"]["inflows"]["values"]:
+                d = d["waterAnalysisInputs"]
```

Comment:
why did you remove the temperature/pressure catch here? If it is removed, then the relative/absolute value boolean for the survey definitely needs to be multi-dimensional.

If a user wants to sweep across temperature (more likely to be absolute) and concentration (more likely to be relative), how would this work? If T/P are handled explicitly as previously, we could maybe avoid that issue. As it is now, we definitely need to deal with it.

Reply:
Makes sense - I'll reimplement the catch for T/P and also pH.